src/store/auth/reducer.tsx

```diff
@@ -1,56 +1,95 @@
 import produce from 'immer';
 import { Action, createReducer } from 'typesafe-actions';
 
-import { AuthState, AuthUser } from '../../components/Auth/types';
+import {
+  AuthProviderInfo,
+  AuthState,
+  AuthUser,
+} from '../../components/Auth/types';
 import { getUniqueId } from '../../components/Firestore/DocumentEditor/utils';
+import { mapResult } from '../utils';
 import * as authActions from './actions';
 
-const INIT_STATE = { users: [], filter: '' };
+const INIT_STATE = {
+  users: { loading: true },
+  filter: '',
+  allowDuplicateEmails: false,
+};
+
+function populateProviderUserInfo(user: AuthUser): AuthUser {
```

Comment:
We really should just refetch the user after updating, since the API already does this for us.

Reply:
Not yet done, but we can tackle this later